src/js/components/homepage/visualizations/treemap/BudgetSubfunctions/BudgetSubfunctionsMap.jsx

```diff
@@ -0,0 +1,237 @@
+/**
+ * BudgetSubfunctionsMap.jsx
+ * Created by Emily Gullo 03/15/2017
+ **/
+
+import React from 'react';
+import { hierarchy, treemap, treemapBinary, treemapSlice } from 'd3-hierarchy';
+import _ from 'lodash';
+import * as MoneyFormatter from 'helpers/moneyFormatter';
+
+import BudgetFunctionCell from '../BudgetFunctions/BudgetFunctionCell';
+import TreeMapTooltip from '../TreeMapTooltip';
+
+const propTypes = {
+    category: React.PropTypes.object,
+    colors: React.PropTypes.array,
+    subfunction: React.PropTypes.object,
+    tooltipStyles: React.PropTypes.object
+};
+
+export default class BudgetSubfunctionsMap extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            windowWidth: 0,
+            visualizationWidth: 0,
+            visualizationHeight: 286,
+            finalNodes: [],
+            hoveredFunction: -1,
+            negativeSubfunctions: []
+        };
+
+        this.handleWindowResize = _.throttle(this.handleWindowResize.bind(this), 50);
+        this.buildTree = this.buildTree.bind(this);
+        this.createTooltip = this.createTooltip.bind(this);
+        this.toggleTooltipIn = this.toggleTooltipIn.bind(this);
+        this.toggleTooltipOut = this.toggleTooltipOut.bind(this);
+    }
+
+    componentDidMount() {
+        this.handleWindowResize();
+        window.addEventListener('resize', this.handleWindowResize);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.subfunction.children.length > 0) {
+            this.buildTree(nextProps);
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleWindowResize);
+    }
+
+    handleWindowResize() {
+        // determine if the width changed
+        const windowWidth = window.innerWidth;
+        if (this.state.windowWidth !== windowWidth) {
+            // width changed, update the visualization width
+            this.setState({
+                windowWidth,
+                visualizationWidth: this.sectionWrapper.offsetWidth
+            });
+            if (this.props.subfunction.children.length > 0) {
+                this.buildTree(this.props);
+            }
+        }
+    }
+
+    calculateTotal(total) {
+        // Remove negative budget subfunction totals from the overall calculation
+        let newTotal = total;
+        const negativeSubfunctions = _.remove(this.props.subfunction.children, (v) => v.value <= 0);
+
+        if (negativeSubfunctions.length > 0) {
+            const negativeTotals = _.sumBy(negativeSubfunctions, 'value');
+            newTotal -= negativeTotals;
+        }
+
+        return newTotal;
+    }
+
+    buildTree(treeProps) {
+        // Remove negative values
+        const positiveSubfunctions = treeProps.subfunction;
+        _.remove(positiveSubfunctions.children, (v) => v.value <= 0);
+
```

Comment:
Why not calculate the total at this point (since the total is the same for all the cells) rather than calling the `calculateTotal` function on every cell - especially since that function has to reiterate through the data to remove negative values each time?